src/Tag/Script.php

```diff
@@ -32,6 +35,34 @@ public function getContent(): string
         return $this->content;
     }
 
+    public function generateNonce(?string $prefix = null): self
+    {
+        return $this->nonce(Html::generateId($prefix ?? ''));
+    }
+
+    /**
+     * @link https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/nonce
+     */
+    public function nonce(?string $nonce): self
+    {
+        $new = clone $this;
+
+        if ($nonce === null || $nonce === '') {
+            unset($new->attributes['nonce']);
+        } else {
+            $new->attributes['nonce'] = $nonce;
+        }
+
+        return $new;
+    }
+
+    public function getNonce(): ?string
```

Comment:
Should this method returns null when nonce attribute is empty string?

Reply:
Will be empty string rendered as empty attribute? If yes, then i think it right return type

Reply:
So this is not consistent with method `nonce()` that unset `nonce` attribute when pass empty string. Let's change `nonce()` method behavior and to allow set nonce as empty string?